lighthouse-core/computed/metrics/timing-summary.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * @license Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Audit = require('../../audits/audit.js');
+const TraceOfTab = require('../trace-of-tab.js');
+const Speedline = require('../speedline.js');
+const FirstContentfulPaint = require('./first-contentful-paint.js');
+const FirstMeaningfulPaint = require('./first-meaningful-paint.js');
+const LargestContentfulPaint = require('./largest-contentful-paint.js');
+const FirstCPUIdle = require('./first-cpu-idle.js');
+const Interactive = require('./interactive.js');
+const SpeedIndex = require('./speed-index.js');
+const EstimatedInputLatency = require('./estimated-input-latency.js');
+const TotalBlockingTime = require('./total-blocking-time.js');
+
+class TimingSummary {
+  /**
+     * @param {LH.Artifacts} artifacts
+     * @param {LH.Audit.Context} context
+     * @return {Promise<{metrics: LH.Artifacts.TimingSummary, debugInfo: Array<Record<string,boolean>>}>}
+     */
+  static async summarize(artifacts, context) {
+    const trace = artifacts.traces[Audit.DEFAULT_PASS];
```

Comment:
we typically like to have the caller submit the proper `trace` and `devtoolsLog` for computed artifacts rather than have the artifact take in *all* artifacts 

this would also eliminate the dependency of this file on `audit.js` which is a little unexpected to go backwards up the chain